receiver/gitproviderreceiver/internal/scraper/githubscraper/helpers_test.go

```diff
@@ -1080,113 +1123,575 @@ func TestGetRepoCVEs(t *testing.T) {
+									},
+								},
+							},
+						},
+					},
+					responseCode: http.StatusOK,
+				},
+				codeScanAlertResponse: codeScanAlertResponse{
+					codeScanAlerts: [][]*github.Alert{
+						{
+							{
+								Rule: &github.Rule{
+									SecuritySeverityLevel: github.String("CRITICAL"),
+								},
+							},
+							{
+								Rule: &github.Rule{
+									SecuritySeverityLevel: github.String("LOW"),
+								},
+							},
+						},
 						{
-							SecurityVulnerability: CVENodeSecurityVulnerability{
-								Severity: "BAD_INPUT",
+							{
+								Rule: &github.Rule{
+									SecuritySeverityLevel: github.String("HIGH"),
+								},
+							},
+							{
+								Rule: &github.Rule{
+									SecuritySeverityLevel: github.String("MEDIUM"),
+								},
 							},
 						},
 					},
+					responseCode: http.StatusOK,
 				},
+			}),
+			expectedMap: map[metadata.AttributeCveSeverity]int64{
+				metadata.AttributeCveSeverityHigh:     2,
+				metadata.AttributeCveSeverityMedium:   2,
+				metadata.AttributeCveSeverityLow:      2,
+				metadata.AttributeCveSeverityCritical: 2,
 			},
-			expected: map[metadata.AttributeCveSeverity]int64{},
+			repo:             "r",
+			org:              "o",
+			expectedCVECount: 8,
+			expectedErr:      nil,
+		},
+		{
+			desc: "TestEmptyInputDepBotAndCodeScanningAlert",
+			server: MockServer(&responses{
+				depBotAlertResponse: depBotAlertResponse{
+					depBotsAlerts: []VulnerabilityAlerts{{}},
+					responseCode:  http.StatusOK,
+				},
+				codeScanAlertResponse: codeScanAlertResponse{
+					codeScanAlerts: [][]*github.Alert{{}},
+					responseCode:   http.StatusOK,
+				},
+			}),
+			expectedMap:      map[metadata.AttributeCveSeverity]int64{},
+			repo:             "r",
+			org:              "o",
+			expectedCVECount: 0,
+			expectedErr:      nil,
 		},
 	}
-
 	for _, tc := range testCases {
 		t.Run(tc.desc, func(t *testing.T) {
-			actual := mapSeverities(tc.input)
-			assert.Equal(t, tc.expected, actual)
+			factory := Factory{}
+			defaultConfig := factory.CreateDefaultConfig()
+			settings := receivertest.NewNopCreateSettings()
+			ghs := newGitHubScraper(context.Background(), settings, defaultConfig.(*Config))
+			server := httptest.NewServer(tc.server)
+			ghs.cfg.GitHubOrg = tc.org
+
+			gClient := graphql.NewClient(server.URL, ghs.client)
+			rClient := github.NewClient(nil)
+
+			url, err := url.Parse(server.URL + "/api-v3" + "/")
+			assert.NoError(t, err)
+			rClient.BaseURL = url
+			rClient.UploadURL = url
+
+			cves, err := ghs.getCVEs(context.Background(), gClient, rClient, tc.repo)
+			totalCVEs := int64(0)
+
+			for _, sevCount := range cves {
+				totalCVEs += sevCount // Assuming 'cves' is a slice of VulnerabilityAlerts
+			}
+
+			assert.Equal(t, tc.expectedCVECount, totalCVEs)
+
+			if tc.expectedErr == nil {
+				assert.NoError(t, err)
+			} else {
+				assert.EqualError(t, err, tc.expectedErr.Error())
+			}
 		})
 	}
 }
+
+// func delete(t *testing.T) {
```

Comment:
let's delete this if we're not running this code.

Reply:
Also deleted, marking as resolved